src/igl/vulkan/ResourcesBinder.cpp

```diff
@@ -78,7 +78,9 @@ void ResourcesBinder::bindSamplerState(uint32_t index, igl::vulkan::SamplerState
   igl::vulkan::VulkanSampler* newSampler = samplerState ? ctx_.samplers_.get(samplerState->sampler_)
                                                         : nullptr;
 
-  if (bindingsTextures_.samplers[index] != newSampler->vkSampler) {
+  IGL_DEBUG_ASSERT(newSampler);                                                      
```

Comment:
A null sampler should be a valid call to `bindSamplerState()`.

I have a proper fix for this, it should appear on GitHub shortly.

Reply:
OK，I'll wait for your new modifications.